test.py

```diff
@@ -19,41 +19,236 @@
 import os
 import subprocess
 import sys
+import time
 
+import sequence_data_processing
+from model_building.predictor import Predictor
 
 def main():
     """
     Script used to perform regression test of the library
 
     This script is used to self check the library.
+    #TODO: aggiorna la prossima riga
     It runs all the examples of example_configurations and checks if there is any error during their execution. Quality of the results is not analyzed nor compared with any reference
     """
-
     parser = argparse.ArgumentParser(description="Performs regression tests")
     parser.add_argument('-d', "--debug", help="Enable debug messages", default=False, action="store_true")
-    parser.add_argument('-j', help="The number of processes to be used", default=1)
+    parser.add_argument('-o', "--output", help="output folder where all the models will be stored", default="output_test")
     args = parser.parse_args()
 
-    # The absolute path of the current script
-    abs_script = os.path.abspath(sys.argv[0])
-
-    # The root directory of the script
-    abs_root = os.path.dirname(abs_script)
-
-    for file in os.listdir(os.path.join(abs_root, "example_configurations")):
-        if not file.endswith(".ini"):
-            continue
-        extra_options = ""
-        if args.debug:
-            extra_options = extra_options + " -d"
-        extra_options = extra_options + " -j" + str(args.j)
-        command = os.path.join(abs_root, "run.py") + " -l -t -c " + os.path.join(abs_root, "example_configurations", file) + " -o output_" + file + extra_options
-        print("Running " + command)
-        ret_program = subprocess.call(command, shell=True, executable="/bin/bash")
-        if ret_program:
-            print("Error in running " + file)
+    done_file_flag = os.path.join(args.output, 'done')
+
+    try:
+        os.mkdir(args.output)
+    except FileExistsError:
+        if os.path.exists(done_file_flag):
+            print(args.output+" already exists. Terminating the program...")
             sys.exit(1)
+        
+
+    tests = generate_tests() #TODO: documenta variabile tests (nel senso, controlla se/come documentarla)
```

Comment:
I don't think any documentation is needed here, one can just read the `generate_tests()` function